pkg/apis/submariner.io/v1/types.go

```diff
@@ -175,3 +175,100 @@ func (c *Connection) SetStatus(status ConnectionStatus, messageFormat string, a
 	c.Status = status
 	c.StatusMessage = fmt.Sprintf(messageFormat, a...)
 }
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GlobalEgressIP defines a policy for allocating GlobalIPs for selected pods in the namespace of the GlobalEgressIP object.
+type GlobalEgressIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the desired behavior.
+	Spec GlobalEgressIPSpec `json:"spec"`
+
+	// The most recently observed status. Read-only.
+	// +optional
+	Status GlobalEgressIPStatus `json:"status,omitempty"`
+}
+
+type GlobalIPAllocationSpec struct {
+	// The requested number of contiguous GlobalIPs to allocate from the Globalnet CIDR assigned to the cluster.
+	// If not specified, defaults to 1.
+	// +optional
+	NumGlobalIPs *int `json:"numGlobalIPs,omitempty"`
+}
+
+type GlobalEgressIPSpec struct {
+	// Specifies how GlobalIPs should be allocated.
+	// +optional
+	Allocation GlobalIPAllocationSpec `json:"allocation,omitempty"`
+
+	// Selects specific pods in the namespace of this GlobalEgressIP to which this GlobalEgressIP applies. If not specified,
+	// all pods in the namespace are selected.
+	// If a pod matches multiple GlobalEgressIP objects, there is no guarantee from which GlobalEgressIP its
+	// GlobalIP will be assigned.
+	// +optional
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
+}
+
+type GlobalEgressIPConditionType string
+
+const (
+	GlobalEgressIPAllocated GlobalEgressIPConditionType = "Allocated"
+	GlobalEgressIPConflict  GlobalEgressIPConditionType = "Conflict"
```

Comment:
Since user is not allowed to explicitly specify the `GlobalIP`, IMHO we do not require "Conflict". But we will require "Error" to specify that Globalnet was unable to allocate GlobalIP for certain reason.

Reply:
`metav1.Condition` also has a `ConditionStatus` field defined as:

```
ConditionTrue    ConditionStatus = "True"
ConditionFalse   ConditionStatus = "False"
ConditionUnknown ConditionStatus = "Unknown"
```

Type defines a condition and ConditionStatus tells its status wrt that condition. So an error would have type "Allocated"  with ConditionStatus "False" and Reason/Message set appropriately.  

"Conflict" could be used if ClusterGlobalEgressIP is a singleton but the user created another one. Not sure if we'll need this type but doesn't hurt to define it.

Reply:
Okay cool. Looks good then.

Reply:
Normally we use the term, globalIP allocation. Personally, I'd prefer "Allocated" over "Valid"